runtime/compiler/trj9/control/CompilationThread.cpp

```diff
@@ -10305,23 +10326,23 @@ TR::CompilationInfo::computeFreePhysicalMemory(bool &incompleteInfo)
 
             if (!incomplete)
                {
-               if (cached == 0 || availSwap == 0)
+               if (cached == 0)
```

Comment:
We should distinguish between not having the information (thus incomplete) and the data being 0. If the portlib returns 0 for memInfo.cached or memInfo.buffered, then we wrongly assume that we have incomplete information.
It's better to do 
```
if (memInfo.cached == OMRPORT_MEMINFO_NOT_AVAILABLE)
   {
   incomplete = true;
   cached = 0;
   }
```
and similarly for buffered.

Reply:
We did not consider the case when cached or buffered later becomes available and we need to set incompete to false.
```
if (!incomplete)
               {
               if (memInfo.cached == OMRPORT_MEMINFO_NOT_AVAILABLE)
                  incomplete = true;
#if defined(LINUX)
               else if (memInfo.buffered == OMRPORT_MEMINFO_NOT_AVAILABLE)
                  incomplete = true;
#endif
               else
                  incomplete = false;
               }
 ```
Again, I'm a bit confused about this, if either cached or buffered was found unavailable, does it mean it's unavailable till the end of the program?

Reply:
In my mind, if the portlib responds with `OMRPORT_MEMINFO_NOT_AVAILABLE` it means it cannot get that information from the OS. I don't see why later on we could get that information. Either the OS won't provide it or it will (including a value of 0) and this should remain consistent throughout the life of the JVM.

Reply:
If it remains consistent throughout the life of the JVM, then I dont' think we need a `static bool incomplete` variable and the pass by reference parameter of `bool &incompleteInfo` at all. We could just set `_cachedFreePhysicalMemoryB` to `OMRPORT_MEMINFO_NOT_AVAILABLE` as soon as we detect that any of the components is `OMRPORT_MEMINFO_NOT_AVAILABLE`. And this is if we have decided that we are never going to print when the reported physical memory data is incomplete.

Reply:
I agree that `static bool incomplete` is not needed. However, the parameter `bool &incompleteInfo` is still needed just in case someone else is interested in partial information.